src/server.py

```diff
@@ -0,0 +1,162 @@
+"""Server programmed to recieve messages from client and return http response."""
+
+
+import socket
+import sys
+
+
+def server():
+    """Recieve a message from the client and sends a response back."""
+    while True:
+        server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_TCP)
+        address = ("127.0.0.1", 7000)
+        server_socket.bind(address)
+        server_socket.listen(1)
+        conn, addr = server_socket.accept()
+        buffer_length = 8
+        req = buffer_request(buffer_length, conn)
+        print(req)
+        conn.sendall(parse_request(req).encode('utf-8'))
+        conn.close()
+
+
+def buffer_request(buffer_length, conn):
+    """Return a decoded client request."""
+    req = ""
+    while True:
+        addition = conn.recv(buffer_length).decode('utf-8')
+        req += addition
+        if len(addition) < buffer_length:
+            break
+    return req
+
+
+def parse_request(request):
+    """Return a server response based on client request formatting."""
+    if not method_validation(request):
+        return response_error("method")
+    elif not version_validation(request):
+        return response_error("version")
+    elif not host_validation(request):
+        return response_error("host")
+    elif not format_validation(request):
+        return response_error("format")
+    return response_error("OK")
+
+
+def method_validation(request):
+    """Verify method formatting and type."""
+    if request[0:4] != "GET ":
+        return False
+    return True
+
+
+def version_validation(request):
+    """Verify version formatting and type."""
+    for ind in range(0, len(request)):
```

Comment:
Why do you need to iterate through the whole request until you find the thing you're looking for? The first line of the request header is always `<METHOD> <URI> <VERSION>`. Why not just split on spaces and take the value at index 2?